logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/e2ei/usecase/GetE2EICertificateUseCase.kt

```diff
@@ -37,10 +36,7 @@ class GetE2eiCertificateUseCaseImpl internal constructor(
 ) : GetE2eiCertificateUseCase {
     override suspend operator fun invoke(clientId: ClientId): GetE2EICertificateUseCaseResult =
         mlsConversationRepository.getClientIdentity(clientId).fold(
-            {
-                if (it is StorageFailure.DataNotFound) GetE2EICertificateUseCaseResult.NotActivated
-                else GetE2EICertificateUseCaseResult.Failure
```

Comment:
are we sure this will fix it but not break other stuff
what if the self conversation is there but the cert is actually `DataNotFound`

Reply:
I'm quit sure about it. 
`DataNotFound` is a child of `StorageFailure`, MLS shouldn't return it. 
If user has no certificate `mlsConversationRepository.getClientIdentity` returns `Either.Right(null)`.